main/routines.c

```diff
@@ -545,10 +555,24 @@ extern const char *baseFilename (const char *const filePath)
 	 */
 	for (i = 0  ;  i < strlen (PathDelimiters)  ;  ++i)
 	{
+#ifdef HAVE_MBLEN
+		const char *p;
+		int ml;
+
+		for (p = filePath  ;  *p != '\0'  ;  ++p)
+		{
+			ml = mblen(p, MB_LEN_MAX);
+			if (ml > 1)
+				p += ml - 1;
+			else if (*p == PathDelimiters [i] && p > tail)
+				tail = p;
+		}
+#else
```

Comment:
First, this code path has nothing to do with `--encoding` AFAICT.

Then, well, as said if we wanted to do it right, we should find the path encoding (basing this on the locale is *somewhat* correct) and convert it to e.g. UTF-8, which we can trust.  What is here is IMO a hack at best.

If we really want to keep this code, I'd say we need at least a comment explaining why it's OK to do this wrong for the designed purpose (so someone like me won't wonder what the heck is this :)